db/new_seeds/scenarios/school_cohorts/fip.rb

```diff
@@ -29,12 +29,43 @@ def with_programme(**args)
         #   unless partnership: :none received. Default: create a new one.
         # Optionally sets it as the default induction programme of the school cohort. Default: true.
         def add_programme(default_induction_programme: true, partnership: nil)
+          supplied_partnership = partnership
           NewSeeds::Scenarios::InductionProgrammes::Fip.new(school_cohort:)
                                                        .build(default_induction_programme:)
-                                                       .with_partnership(partnership:)
+                                                       .with_partnership(partnership: supplied_partnership || @partnership)
                                                        .induction_programme
         end
 
+        def with_partnership(**args)
+          and_partnership(**args)
+
+          self
+        end
+
+        def and_partnership(lead_provider: nil, delivery_partner: nil)
+          traits = []
+          traits << :with_lead_provider if lead_provider.nil?
+          traits << :with_delivery_partner if delivery_partner.nil?
+
+          options = {
+            lead_provider:,
+            delivery_partner:,
+            cohort:,
+            school:,
+          }.compact
+
+          @partnership = FactoryBot.create(:seed_partnership, *traits, **options)
+
+          FactoryBot.create(
+            :seed_provider_relationship,
+            cohort:,
+            lead_provider: partnership.lead_provider,
+            delivery_partner: partnership.delivery_partner,
+          )
+
+          @partnership
```

Comment:
Should this be returning `self` so it's chainable?

Reply:
from what I saw in others the pattern was that "with_XXX" is chainable and "add_XXX" returns the thing that has been added?